storage/innobase/row/row0mysql.cc

```diff
@@ -5500,6 +5500,19 @@ row_rename_table_for_mysql(
 		goto funct_exit;
 	}
 
+
+	/* Wait for background fts sync to finish */
+	for (retry = 1; dict_fts_index_syncing(table); ++retry) {
```

Comment:
I don't quite understand how the accesses to dict_index_t::index_fts_syncing itself are synchronized - the writes are protected by fts_cache_t::lock, and reads by nothing? If so, that looks questionable, but not our job to fix

Reply:
Right, I don't like this variable and sync mechanism. May be a rw_lock or MDL lock on table should be used. The changes could be huge and will differ from upstream syncing logic.